docs/book/getting-started/installation/m1-mac-installation.md

```diff
@@ -0,0 +1,228 @@
+
+### Verifying whether you run under Rosetta
+
+To check whether your terminal is running under Rosetta or not, use the `arch`
+command:
+
+```
+arch
+```
+
+This should return `i386` if you are running under Rosetta.
+
+### Using Rosetta Terminal in VSCode
+
+If you are using VSCode, you can add a Rosetta terminal option by modifying
+your `settings.json`. 
+To do so, press `CMD + Shift + P`, search for 'Preferences: Open User Settings 
+(JSON)', then add an entry into `terminal.integrated.profiles.osx` as shown
+below.
+
+```
+"terminal.integrated.profiles.osx": {
+    ...
+    "rosetta": {
+        "path": "arch",
+        "args": ["-x86_64", "zsh", "-l"],
+        "overrideName": true
+    }
+},
+```
+
+Optionally, you can also set Rosetta ZSH as the default terminal by setting
+`terminal.integrated.defaultProfile.osx` accordingly:
+
+```
+"terminal.integrated.defaultProfile.osx": "rosetta",
+```
+
+
+## 3. Installing Homebrew under Rosetta
+
+Next, we will install Homebrew under Rosetta and alias it to `brew86`.
+
+To install Homebrew under Rosetta, use the following command:
+
+```
+arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+```
+
+Then, create the `brew86` alias like so:
+
+```
+alias brew86="arch -x86_64 /usr/local/bin/brew"
+```
+
+## 4. Installing pyenv and pyenv-virtualenv under Rosetta
+
+### Installing Dependencies
+
+To install pyenv, you first need to install its dependencies:
+
+```
+brew86 install openssl readline sqlite3 xz zlib tcl-tk
+```
+
+### Installing pyenv
+
+Then you can install pyenv under Rosetta via:
+
+```
+brew86 install pyenv
+```
+
+### Installing pyenv-virtualenv
+
+Similarly, you can install `pyenv-virtualenv` via:
+
+```
+brew86 install pyenv-virtualenv
+```
+
+### Creating a pyenv86 Alias
+
+Then we will again create an alias for running pyenv under Rosetta:
+
+```
+alias pyenv86="arch -x86_64 pyenv"
+```
+
+### Setting Environment Variables
+
+Lastly, run the following lines to set the environment variables required by
+pyenv and pyenv-virtualenv in your `.zshrc`:
+
+```
+echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.zshrc
+echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.zshrc
+echo 'eval "$(pyenv init -)"' >> ~/.zshrc
+echo 'eval "$(pyenv virtualenv-init -)"' >> ~/.zshrc
+```
+
+## 5. Installing Python under Rosetta with Pyenv
+
+Now you're all set and can install the Python version of your choice via
+
+```
+pyenv86 install <PYTHON_VERSION>
+```
+
+E.g. to install the latest Python version that ZenML currently supports (3.9.14),
+run:
+
+```
+pyenv86 install 3.9.14
+```
+
+## 6. Installing ZenML in a Virtual Environment
+
+Lastly, let's install ZenML in a new virtual environment.
+
+### Creating a Virtual Environment with pyenv-virtualenv
+
+To create a new virtual environment called `zenml` based on Python 3.9.14, run:
+
+```
+pyenv virtualenv 3.9.14 zenml
```

Comment:
```suggestion
pyenv86 virtualenv 3.9.14 zenml
```
I guess?

Reply:
Yes! Thanks for spotting this :)

Reply:
Oh, no, actually you just need `pyenv86` for installation of Python versions, creating venvs and settings them active works with both versions.